### PR TITLE
Update the conda environment to create the documentation locally

### PR DESCRIPTION
**Pull request by steffengraber**
This PR updates the file for creating the conda environment to build the documentation locally. The corresponding README file is updated accordingly.

### CODE REVIEW

**Comment by heplesser** [comment]
Escalated to "high", since #1439 is blocked by this one.

File: doc/conda_environment.yml [hunk 1]
```diff
@@ -1,73 +1,51 @@
 name: doc
```
[the remark this answers is not in the document]
Comment (steffengraber) [reply doc/conda_environment.yml]: Very good idea. It' done.

**Comment by sarakonradi** [comment]
> @sarakonradi Would you adjust the workflow documentation in #1439 to these name changes?

Done.

**Comment by heplesser** [comment]
Since this PR only addresses details on local building of documentation relevant for developers, I merge without further review.

**Pull request merged**